codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/endpoint/EndpointsDecorator.kt

```diff
@@ -43,18 +43,28 @@ interface EndpointCustomization {
      * Provide the default value for [parameter] given a reference to the service config struct ([configRef])
      *
      * If this parameter is not recognized, return null.
+     *
+     * Example:
+     * ```kotlin
+     * override fun loadBuiltInFromServiceConfig(parameter: Parameter, configRef: String): Writable? {
+     *     return when (parameter.builtIn) {
+     *         Builtins.REGION.builtIn -> writable { rust("$configRef.region.as_ref().map(|r|r.as_ref().to_owned())") }
+     *         else -> null
+     *     }
+     * }
+     * ```
      */
-    fun builtInDefaultValue(parameter: Parameter, configRef: String): Writable? = null
+    fun loadBuiltInFromServiceConfig(parameter: Parameter, configRef: String): Writable? = null
 
     /**
-     * Provide a list of additional endpoints standard library functions that rules can use
+     * Set a given builtIn value on the service config builder. If this builtIn is not recognized, return null
      */
-    fun customRuntimeFunctions(codegenContext: ClientCodegenContext): List<CustomRuntimeFunction> = listOf()
+    fun setBuiltInOnServiceConfig(name: String, value: Node, configBuilderRef: String): Writable? = null
 
     /**
-     * Set a given builtIn value on the service config builder. If this builtIn is not recognized, return null
+     * Provide a list of additional endpoints standard library functions that rules can use
      */
-    fun setBuiltInOnConfig(name: String, value: Node, configBuilderRef: String): Writable? = null
+    fun customRuntimeFunctions(codegenContext: ClientCodegenContext): List<CustomRuntimeFunction> = listOf()
```

Comment:
I appreciate the example on `loadBuiltInFromServiceConfig`. Adding examples for these two would be just as appreciated.